### PR TITLE
feat: Submit deep link via window

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: electron/src/lib/CoreProtocol.ts [hunk 1]
```diff
@@ -93,26 +93,50 @@ export class CustomProtocolHandler {
...
+          logger.info('No deep link found in arguments.');
         }
       });
       app.on('second-instance', async (_event, argv) => {
         logger.info('Second instance detected, looking for deep link in arguments ...');
         const deepLink = this.findDeepLink(argv);
         if (deepLink) {
-          logger.info('Second instance detected, dispatching deep link ...');
-          await this.dispatchDeepLink(deepLink);
+          logger.info('Dispatching deep link ...');
```
Comment (lipis) [inline electron/src/lib/CoreProtocol.ts:132]:
```suggestion
          logger.info('Dispatching deep link...');
```

File: electron/src/lib/CoreProtocol.ts [hunk 2]
```diff
@@ -93,26 +93,50 @@ export class CustomProtocolHandler {
...
+        }
       });
     } else {
       app.once('ready', async () => {
         logger.info('App ready, looking for deep link in arguments ...');
         const deepLink = this.findDeepLink(process.argv);
         if (deepLink) {
-          logger.info('App ready, dispatching deep link ...');
-          await this.dispatchDeepLink(deepLink);
+          logger.info('Dispatching deep link ...');
```
Comment (lipis) [inline electron/src/lib/CoreProtocol.ts:118]:
```suggestion
          logger.info('Dispatching deep link...');
```

**Review by lipis — APPROVED** [review]

File: electron/src/lib/CoreProtocol.ts [hunk 3]
```diff
@@ -93,26 +93,50 @@ export class CustomProtocolHandler {
...
-          logger.info('Second instance detected, dispatching deep link ...');
-          await this.dispatchDeepLink(deepLink);
+          logger.info('Dispatching deep link ...');
+          try {
+            await this.dispatchDeepLink(deepLink);
+          } catch (error) {
+            logger.error(error);
+          }
+        } else {
+          logger.info('No deep link found in arguments.');
```
Comment (thisisamir98) [inline electron/src/lib/CoreProtocol.ts:139]: @ffflorian you have 4 try catch blocks here which all are doing the same thing, maybe you could extract them to one function?

File: electron/src/lib/CoreProtocol.ts (hunk 1 above)
Comment (ffflorian) [reply electron/src/lib/CoreProtocol.ts:132]: We are using a space before ellipses in all the other places, too. I am going to keep the space for consistency.

File: electron/src/lib/CoreProtocol.ts (hunk 2 above)
Comment (ffflorian) [reply electron/src/lib/CoreProtocol.ts:118]: We are using a space before ellipses in all the other places, too. I am going to keep the space for consistency.

File: electron/src/lib/CoreProtocol.ts [hunk 4]
```diff
@@ -93,26 +94,34 @@ export class CustomProtocolHandler {
...
+
     if (platform.IS_MAC_OS) {
       app.on('open-url', async (event, url) => {
         event.preventDefault();
-        await this.dispatchDeepLink(url);
+        try {
+          await this.dispatchDeepLink(url);
+        } catch (error) {
+          logger.error(error);
+        }
```
Comment (thisisamir98) [inline electron/src/lib/CoreProtocol.ts:106]: I think you forgot to use forwardDeepLink here

**Review by thisisamir98 — APPROVED** [review]

**Pull request merged**